app/V1Module/presenters/ExercisesPresenter.php

```diff
@@ -694,6 +694,75 @@ public function actionAllTags() {
     $this->sendSuccessResponse($tags);
   }
 
+  public function checkTagsStats() {
+    if (!$this->exerciseAcl->canViewTagsStats()) {
+      throw new ForbiddenRequestException("You are not allowed to view tags statistics");
+    }
+  }
+
+  /**
+   * Get list of global exercise tag names along with how many times they have been used.
+   * @GET
+   */
+  public function actionTagsStats() {
+    $stats = $this->exerciseTags->getStatistics();
+    $this->sendSuccessResponse($stats);
+  }
+
+  public function checkTagsUpdateGlobal(string $tag) {
+    if (!$this->exerciseAcl->canUpdateTagsGlobal()) {
+      throw new ForbiddenRequestException("You are not allowed to update tags globally");
+    }
+  }
+
+  /**
+   * Update the tag globally. At the moment, the 'update' means 'rename' as there is no other relevant function available.
+   * @POST
+   * @param string $tag Tag to be updated
+   * @Param(type="query", name="renameTo", validation="string:1..32", required=false, description="New name of the tag")
+   * @Param(type="query", name="force", validation="bool", required=false, description="If true, the rename will be allowed even if the new tag name exists (tags will be merged). Otherwise, name collisions will result in error.")
+   */
+  public function actionTagsUpdateGlobal(string $tag, string $renameTo = null, bool $force = false) {
+    if ($renameTo === null) {
```

Comment:
This seems a bit weird, `renameTo` is flagged as non-required, yet here lies the check to emptiness and below is bad request response...